core/vil1/vil1_stream_url.cxx

```diff
@@ -226,10 +226,10 @@ vil1_stream_url::vil1_stream_url(char const * url)
   char buffer[4096];
 
   // send HTTP 1.1 request.
-  std::sprintf(buffer, "GET /%s / HTTP/1.1\n", path.c_str());
+  std::snprintf(buffer, sizeof(buffer), "GET /%s / HTTP/1.1\n", path.c_str());
   if (!auth.empty())
-    std::sprintf(buffer + std::strlen(buffer), "Authorization:  Basic %s\n", encode_base64(auth).c_str());
-    //    std::sprintf(buffer+std::strlen(buffer), "Authorization:  user  testuser:testuser\n");
+    std::snprintf(buffer + std::strlen(buffer), sizeof(buffer) - std::strlen(buffer) - 1 , "Authorization:  Basic %s\n", encode_base64(auth).c_str());
+    //    std::snprintf(buffer+std::strlen(buffer), sizeof(buffer) "Authorization:  user  testuser:testuser\n");
```

Comment:
Looking at this again, it's not clear 4096 is even big enough, so then you'd have to check the return value of snprintf, and then it gets all ugly.

I'd remove this from this 'trivial' PR and rewrite this bit to use `std::string`...